### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by frush86**

### CODE REVIEW

**Comment by cbuerger92** [comment]
🥳

**Review by Thildi — COMMENTED** [review]
VERY GOOOOOD

File: index.js [hunk 1]
```diff
@@ -11,26 +11,58 @@ const nextButton = document.querySelector('[data-js="button-next"]');
...
+let maxPage = 1; // changed to let
+let page = 1;
 const searchQuery = "";
 
-async function fetchCharacters() {
-  const response = await fetch("https://rickandmortyapi.com/api/character");
-  const data = await response.json();
-
-  return data.results;
+async function fetchCharacters(page) {
```
Comment (Roland-Hufnagel) [inline index.js:18]: You don't need the parameter here. Your function can access the global variable 'page'.

File: index.js [hunk 2]
```diff
@@ -11,26 +11,58 @@ const nextButton = document.querySelector('[data-js="button-next"]');
...
+  try {
```
Comment (Roland-Hufnagel) [inline index.js:19]: The first one with error handling!!! Love it ❤️

File: index.js [hunk 3]
```diff
@@ -11,26 +11,58 @@ const nextButton = document.querySelector('[data-js="button-next"]');
...
+    );
+    if (!response.ok) {
+      throw new Error(`status:${response.status}`);
+    }
+    const data = await response.json();
+    maxPage = data.info.pages; //
+    console.log(data);
+    return data.results;
+  } catch (error) {
+    console.log("Error!!!");
```
Comment (Roland-Hufnagel) [inline index.js:31]: Since you are creating an explicit error message in line 24 you can use it here:
```console.log(`Error, status: ${error.message}`)```

**Review by Roland-Hufnagel — COMMENTED** [review]
Great 💪. Only a few suggestions...

**Pull request merged**